### PR TITLE
wrap data-api-client in a namespace to improve type visibility

### PR DESCRIPTION
**Pull request by hariseldon78**
I had the necessity to use the type iDataAPIClient in my own interface, but with the current declaration I couldn't import that type in any way. So I edited the declaration in this way, putting everything inside a namespace which allow me to do stuff like this:
```
import Client from 'data-api-client';

export async function inTransaction(
    f: (dbClient: Client.iDataAPIClient, transactionId: string) => Promise<void>
) {
    const secretArn = process.env.AURORA_SECRET_ARN;
    const resourceArn = process.env.AURORA_CLUSTER_ARN;
    const database = process.env.DB_NAME;
    const dbClient = Client({
        secretArn,
        resourceArn,
        database,
    });

    const { transactionId } = await dbClient.beginTransaction();
    if (!transactionId) throw new Error('unable to begin transaction');
    try {
        await f(dbClient, transactionId);
    } catch (e) {
        await dbClient.rollbackTransaction({ transactionId });
        throw e;
    }
    await dbClient.commitTransaction({ transactionId });
}
```

Specifically, in this example, it allows to have a type safe signature for the passed function.

### CODE REVIEW

File: types/data-api-client/tsconfig.json [hunk 1]
```diff
@@ -6,6 +6,7 @@
         "noImplicitThis": true,
         "strictFunctionTypes": true,
         "strictNullChecks": true,
+        "esModuleInterop": true,
```
Comment (amcasey) [inline types/data-api-client/tsconfig.json:9]: Why was this needed?

**Review by amcasey — CHANGES_REQUESTED** [review]

File: types/data-api-client/tsconfig.json (hunk 1 above)
Comment (hariseldon78) [reply types/data-api-client/tsconfig.json:9]: probably not, sorry

**Review by amcasey — APPROVED** [review]
Thanks!

**Comment by hariseldon78** [comment]
Ready to merge
